feat(email): validate email

Kalpana057 opened the pull request:
<!--

name: "Monorepo Contribution" about: "Template for contributions in a monorepo
with apps and packages folders" title: "[Feature/Bug] - Title" labels:
["contribution"] assignees: "" # Leave blank if no specific assignee

---

-->

**Title**: [Feature] - Email Verification UI
**Labels**: contribution

# Description

Created the Email Verification UI as per the design in Figma for the Wallet Management System. The component informs the user to check their email for a password reset link and allows them to resend it or go back to the login screen.

**Fixes**: # 265
or  
**Resolves**: # 265

---

## Changes Made

- [x] Changes in **`apps`** folder (specify the app and briefly describe the
      changes): 

  - [x] `Web`  Added VerifyEmail component implementing the "Check your email" screen from Figma
  - [ ] `Native`

- [ ] Changes in **`packages`** folder (specify the package and briefly describe
      the changes):
  - [ ] `Core`

---

### Type of Change

- [ ] 🐛 **Bug fix** (non-breaking change which fixes an issue)
- [x] ✨ **New feature** (non-breaking change which adds functionality)
- [ ] 💥 **Breaking change** (fix or feature that would cause existing
      functionality to not work as expected)
- [ ] 📝 **Documentation update** (changes)

---

#### Screenshots

|       Before        |       After        |
| :-----------------: | :----------------: |
| "screenshot before" | ![image](https://github.com/user-attachments/assets/1085bed1-3fe1-41f8-9b6d-c5673ae9e6fb)|

---

### How Has This Been Tested?
- [x] Manually tested component rendering 
- [ ] Cypress integration
- [ ] Cypress component tests
- [ ] Jest unit tests

---

### Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

---

### Additional Comments

_(Optional) Add any additional comments or notes for reviewers here._

pierrelstan reviewed (changes requested):
You import Box, Typography, and Button from @mui/material twice (once as a group, once for IconButton).
Suggestion: Import all MUI components from a single import statement.

Kalpana057 commented:
Hi @pierrelstan 

I have removed the duplicated imports and added single import statement

I have added aria-label for "Back to login" accessibility.

Let me know if any other changes need to be done
